File: app/src/main/java/com/jakebarnby/filemanager/sources/googledrive/GoogleDriveFile.java
package com.jakebarnby.filemanager.sources.googledrive;


import com.google.api.services.drive.model.File;
import com.jakebarnby.filemanager.sources.models.SourceFile;
import com.jakebarnby.filemanager.sources.models.SourceType;
import com.jakebarnby.filemanager.util.Constants;

/**
 * Created by Jake on 6/5/2017.
 */

public class GoogleDriveFile extends SourceFile {

    private String mDriveId;

    public GoogleDriveFile(File file) {
        if (file.getWebContentLink() != null) {
            setPath(file.getWebContentLink());
        }
        setDriveId(file.getId());
        setName(file.getName());
        setSourceType(SourceType.REMOTE);
        setSourceName(Constants.Sources.GOOGLE_DRIVE);
        setDriveId(file.getId());
        setDirectory(file.getMimeType().equals(Constants.Sources.GOOGLE_DRIVE_FOLDER_MIME));
        setThumbnailLink(file.getHasThumbnail() ? file.getThumbnailLink() : file.getIconLink());
        setModifiedTime(file.getModifiedTime().getValue());
        setHidden(file.getName().startsWith("."));

        if (!isDirectory()) {
            setSize(file.getSize() == null? 0 : file.getSize());
        }

        if (file.getWebViewLink() != null) {
            setPath(file.getWebViewLink());
        }
    }
    public String getDriveId() {
        return mDriveId;
    }

    public void setDriveId(String mDriveId) {
        this.mDriveId = mDriveId;
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/sources/googledrive/GoogleDriveFile.kt
package com.jakebarnby.filemanager.sources.googledrive

import com.google.api.services.drive.model.File
import com.jakebarnby.filemanager.models.SourceFile
import com.jakebarnby.filemanager.models.SourceType
import com.jakebarnby.filemanager.util.Constants.Sources

/**
 * Created by Jake on 6/5/2017.
 */
class GoogleDriveFile(
    file: File
) : SourceFile() {

    var driveId: String

    init {
        if (file.webContentLink != null) {
            path = file.webContentLink
        }
        if (file.webViewLink != null) {
            path = file.webViewLink
        }

        driveId = file.id
        name = file.name
        sourceId = SourceType.GOOGLE_DRIVE.id
        driveId = file.id
        isDirectory = file.mimeType == Sources.GOOGLE_DRIVE_FOLDER_MIME

        thumbnailLink = if (file.hasThumbnail) {
            file.thumbnailLink
        } else {
            file.iconLink
        }
        modifiedTime = file.modifiedTime.value
        isHidden = file.name.startsWith(".")

        if (!isDirectory) {
            size = if (file.getSize() == null) {
                0
            } else {
                file.getSize()
            }
        }
    }
}